src/utils/pure_pursuit.cpp

```diff
@@ -60,105 +60,109 @@ bool PurePursuit::Path::is_valid() { return this->valid; }
  */
-[[maybe_unused]] Translation2d PurePursuit::get_lookahead(const std::vector<Translation2d>& path, Pose2d robot_loc, double radius){
-  // Default: the end of the path
-  Translation2d target = path.back();
+[[maybe_unused]] Translation2d
+PurePursuit::get_lookahead(const std::vector<Translation2d> &path, Pose2d robot_loc, double radius) {
+    // Default: the end of the path
+    Translation2d target = path.back();
 
-  if (target.distance(robot_loc.translation()) <= radius) {
-    return target;
-  }
-
-  // Check each line segment of the path for potential targets
-  for (int i = 0; i < path.size() - 1; i++) {
-    Translation2d start = path[i];
-    Translation2d end = path[i + 1];
-
-    std::vector<Translation2d> intersections = PurePursuit::line_circle_intersections(robot_loc.translation(), radius, start, end);
-    // Choose the intersection that is closest to the end of the line segment
-    // This prioritizes the closest intersection to the end of the path
-    for (Translation2d intersection : intersections) {
-      if (intersection.distance(end) < target.distance(end)) {
-        target = intersection;
-      }
+    if (target.distance(robot_loc.translation()) <= radius) {
+        return target;
     }
-  }
 
-  return target;
+    // Check each line segment of the path for potential targets
+    for (int i = 0; i < path.size() - 1; i++) {
+        Translation2d start = path[i];
+        Translation2d end = path[i + 1];
+
+        std::vector<Translation2d> intersections =
+          PurePursuit::line_circle_intersections(robot_loc.translation(), radius, start, end);
+        // Choose the intersection that is closest to the end of the line segment
+        // This prioritizes the closest intersection to the end of the path
+        for (Translation2d intersection : intersections) {
+            if (intersection.distance(end) < target.distance(end)) {
+                target = intersection;
+            }
+        }
+    }
+
+    return target;
 }
 
 /**
  Injects points in a path without changing the curvature with a certain spacing.
 */
-[[maybe_unused]] std::vector<Translation2d> PurePursuit::inject_path(const std::vector<Translation2d>& path, double spacing) {
-  std::vector<Translation2d> new_path;
+[[maybe_unused]] std::vector<Translation2d>
+PurePursuit::inject_path(const std::vector<Translation2d> &path, double spacing) {
+    std::vector<Translation2d> new_path;
 
-  // Injecting points for each line segment
-  for (int i = 0; i < path.size() - 1; i++) {
-    Translation2d start = path[i];
-    Translation2d end = path[i + 1];
+    // Injecting points for each line segment
+    for (int i = 0; i < path.size() - 1; i++) {
+        Translation2d start = path[i];
+        Translation2d end = path[i + 1];
 
-    Translation2d diff = end - start;
-    Vector2D vector = Vector2D(diff);
+        Translation2d diff = end - start;
+        Translation2d vector = Translation2d(diff);
```

Comment:
this line basically does nothing since diff is already a Tranlation2D
. can remove/rename